pkg/operator/controller/ingress_status.go

```diff
@@ -24,7 +33,25 @@ func (r *reconciler) syncIngressControllerStatus(deployment *appsv1.Deployment,
 	updated := ic.DeepCopy()
 	updated.Status.AvailableReplicas = deployment.Status.AvailableReplicas
 	updated.Status.Selector = selector.String()
-	updated.Status.Conditions = computeIngressStatusConditions(updated.Status.Conditions, deployment)
+
+	updated.Status.Conditions = []operatorv1.OperatorCondition{}
+	updated.Status.Conditions = append(updated.Status.Conditions, computeIngressStatusConditions(updated.Status.Conditions, deployment)...)
+	updated.Status.Conditions = append(updated.Status.Conditions, r.statusCache.computeLoadBalancerStatus(ic)...)
+
+	for i := range updated.Status.Conditions {
```

Comment:
there's probably an elegant functional approach to this one

Reply:
Not anymore it doesn't, I deleted the usage. Isn't the problem completely generic?

Reply:
Yeah.  It is for the conditions that `ingress_status.go` sets.  It looks like it is for the conditions that `status.go` sets as well.  (For `status.go`, we need to be careful to respect the special requirements for updating `LastTransitionTime` of the `Progressing` condition, but we handle that now by assigning a distinct value to its `Message` for the cases in which we need `LastTransitionTime` to be updated.)  So it looks like we can delete `setIngressLastTransitionTime`, the first parameter of `computeIngressStatusConditions`, and the first parameter of `computeIngressAvailableCondition` (and soon `getIngressAvailableCondition` as well), and we could do the same in `ingress.go` too.

Reply:
Maybe we can implement some of these ideas in a follow-up.